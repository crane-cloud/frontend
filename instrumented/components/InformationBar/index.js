import React, { useState } from "react";
//import RoundAddButton from "../RoundAddButton";
import AppStatus from "../AppStatus";
import PrimaryButton from "../PrimaryButton";
import { ReactComponent as SearchButton } from "../../assets/images/search.svg";
import { ReactComponent as Coin } from "../../assets/images/coin.svg";
import "./InformationBar.css";

const InformationBar = ({
  header,
  buttontext,
  status,
  showBtn,
  btnAction,
  btntype = "new",
  viewAppLink,
  credits,
  showSearchBar,
  placeholder,
  searchAction,
}) => {
  const [Searchword, setSearchword] = useState("");
  const callbackSearchWord = ({ target }) => {
    const { value } = target;
    setSearchword(value);
    searchAction(value);
  };
  return (
    <div className="InformationBar SmallContainer">
      {status ? (
        <div className="InformationBarWithButton">
          <div className="AppUrl">
            <a target="_blank" rel="noopener noreferrer" href={header}>
              {header}
            </a>
          </div>
          <div className="RoundAddButtonWrap">
            <AppStatus appStatus={status} />
          </div>
        </div>
      ) : showSearchBar ? (
        <div className="InformationBarColumnView">
          <div className="InformationBarWithButton">
            <div className="InfoHeader">{header}</div>
            <div className="InfoContent">
              <div className="SearchBar DesktopView">
                <div className="SearchInput">
                  <input
                    type="text"
                    className="searchTerm"
                    name="Searchword"
                    placeholder={placeholder}
                    value={Searchword}
                    onChange={callbackSearchWord}
                  />
                  <SearchButton className="SearchIcon" />
                </div>
              </div>
              <div className="RoundAddButtonWrap">
                {/*<RoundAddButton onClick={btnAction} />*/}
                <PrimaryButton btntype={btntype} onClick={btnAction}>
                  {buttontext}
                </PrimaryButton>
              </div>
            </div>
          </div>
          <div className="InfoContent MobileView">
            <div className="SearchBar ">
              <div className="SearchInput">
                <input
                  type="text"
                  className="searchTerm"
                  name="Searchword"
                  placeholder={placeholder}
                  value={Searchword}
                  onChange={callbackSearchWord}
                />
                <SearchButton className="SearchIcon" />
              </div>
            </div>
          </div>
        </div>
      ) : showBtn ? (
        <div className="InformationBarWithButton">
          <div className="InfoHeader">{header}</div>
          <div className="RoundAddButtonWrap">
            {/*<RoundAddButton onClick={btnAction} />*/}
            <PrimaryButton btntype={btntype} onClick={btnAction}>
              {buttontext}
            </PrimaryButton>
          </div>
        </div>
      ) : viewAppLink ? (
        <div className="InformationBarWithButton">
          <div className="InfoHeader">{header}</div>
          <a href={viewAppLink} rel="noopener noreferrer" target="_blank">
            <PrimaryButton color="primary-outline">Open App</PrimaryButton>
          </a>
        </div>
      ) : credits ? (
        <div className="InformationBarWithButton">
          <div className="InfoHeader">{header}</div>
          {/**appears if user has credits */}
          <div className="CreditsCorner" title="credits">
            {credits > 0 ? credits : 0}
            <Coin />
          </div>
        </div>
      ) : (
        <div className="InfoHeader">{header}</div>
      )}
    </div>
  );
};

export default InformationBar;
