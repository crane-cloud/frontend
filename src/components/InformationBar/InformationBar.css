.InformationBar {
  background: #f8f7e5;
  height: 5rem;
  display: flex;
  align-items: center;
}

.InformationBarWithButton {
  width: 100%;
  display: flex;
  gap: 1rem;
  justify-content: space-between;
}

.InfoHeader {
  display: flex;
  align-items: center;
  font-weight: bold;
  font-size: 1.5rem;
}

.InfoContent{
  display: flex;
  justify-content: flex-end;
  gap: 4rem;
  align-items: center;
}

.ViewAppBtn {
  color: #008ac1;
  background-color: transparent;
  outline: 2px solid #008ac1;
}
.ViewAppBtn:hover {
  color: #fff;
  background-color: #008ac1;
  outline: 2px solid #008ac1;
}

.SearchInput{
  display:flex;
}

.SearchInput input{
  background-color: #EBEAD5;
  color: #000;
  border: 0;
  border-top-left-radius: 4px 2px;
  border-bottom-left-radius: 3px 4px;
  font-size: 1rem;
  padding: 1rem;
  height: 45px;
  width: 300px;
}

.SearchIcon{
  background-color: #EBEAD5;
  padding: 0.42rem;
  border-bottom-right-radius: 6px 2px;
  border-top-right-radius: 3px 4px;
}