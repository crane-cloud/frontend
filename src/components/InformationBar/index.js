import React, { useState } from "react";
import { Link } from "react-router-dom";
//import RoundAddButton from "../RoundAddButton";
import AppStatus from "../AppStatus";
import PrimaryButton from "../PrimaryButton";
import { ReactComponent as SearchButton } from "../../assets/images/search.svg";
import { ReactComponent as Coin } from "../../assets/images/coin.svg";
import "./InformationBar.css";
import { ReactComponent as User } from "../../assets/images/user.svg";
import { ReactComponent as Users } from "../../assets/images/users.svg";
const InformationBar = ({
  header,
  buttontext,
  status,
  showBtn,
  btnAction,
  btntype = "new",
  viewAppLink,
  credits,
  showSearchBar,
  placeholder,
  searchAction,
  adminRoute,
  adminProjects,
  selectedProjects,
  handleTabChange,
  handleSharedProjectsTabChange,
  myProjectsList = [],
  sharedProjectsList = [],
  onFilterSelect,
  viewFilter = "False",
}) => {
  const [Searchword, setSearchword] = useState("");
  const callbackSearchWord = ({ target }) => {
    const { value } = target;
    setSearchword(value);
    searchAction(value);
  };
return (
    <div className="InformationBar SmallContainer">
      {status ? (
        <div className="InformationBarWithButton">
          <div className="AppUrl">
            <a target="_blank" rel="noopener noreferrer" href={header}>
              {header}
            </a>
          </div>
          <div className="RoundAddButtonWrap">
            <AppStatus appStatus={status} />
          </div>
        </div>
      ) : showSearchBar ? (
        <div className="InformationBarColumnView">
          <div className="InformationBarWithButton">
            <div className="InfoHeader">
              {header}
              {viewFilter && (
                <div className="InfoProjectCategories">
                  <button
                    className={
                      selectedProjects === "MyProjects"
                        ? "InfoCurrentTab"
                        : "InfoTab"
                    }
                    onClick={() => handleTabChange("MyProjects")}
                  >
                    <User className="SmallerIcon" />
                    <span>
                      MyProjects{" "}
                      <span title="Projects">{`(${myProjectsList.length})`}</span>
                    </span>
                  </button>
                  <button
                    className={
                      selectedProjects === "SharedProjects"
                        ? "CurrentTab"
                        : "Tab"
                    }
                    onClick={() => handleSharedProjectsTabChange("SharedProjects")}
                  >
                    <span>
                      <Users className="SmallerIcon" />
                    </span>
                    <span>
                      {" "}
                      Shared Projects{" "}
                      <span title="Projects">{`(${sharedProjectsList.length})`}</span>{" "}
                    </span>
                  </button>
                </div>
              )}
            </div>
            <div className="InfoContent">
              <div className="SearchBar DesktopView">
                <div className="SearchInput">
                  <input
                    type="text"
                    className="searchTerm"
                    name="Searchword"
                    placeholder={placeholder}
                    value={Searchword}
                    onChange={callbackSearchWord}
                  />
                  <SearchButton className="SearchIcon" />
                </div>
              </div>
              <div className="ButtonWrap">
                {adminRoute &&
                  (adminProjects ? (
                    <Link
                      to={{
                        pathname: `/projects`,
                      }}
                    >
                      <PrimaryButton color="primary">
                        Admin Projects
                      </PrimaryButton>
                    </Link>
                  ) : (
                    <Link
                      to={{
                        pathname: `/clusters`,
                      }}
                    >
                      <PrimaryButton color="primary">Dashboard</PrimaryButton>
                    </Link>
                  ))}
                <PrimaryButton btntype={btntype} onClick={btnAction}>
                  {buttontext}
                </PrimaryButton>
              </div>
            </div>
          </div>
          <div className="InfoContent MobileView">
            <div className="SearchBar ">
              <div className="SearchInput">
                <input
                  type="text"
                  className="searchTerm"
                  name="Searchword"
                  placeholder={placeholder}
                  value={Searchword}
                  onChange={callbackSearchWord}
                />
                <SearchButton className="SearchIcon" />
              </div>
            </div>
          </div>
        </div>
      ) : showBtn ? (
        <div className="InformationBarWithButton">
          <div className="InfoHeader">{header}</div>
          <div className="ButtonWrap">
            {adminRoute &&
              (adminProjects ? (
                <Link
                  to={{
                    pathname: `/projects`,
                  }}
                >
                  <PrimaryButton color="primary">Admin Projects</PrimaryButton>
                </Link>
              ) : (
                <Link
                  to={{
                    pathname: `/clusters`,
                  }}
                >
                  <PrimaryButton color="primary">Dashboard</PrimaryButton>
                </Link>
              ))}
            <PrimaryButton btntype={btntype} onClick={btnAction}>
              {buttontext}
            </PrimaryButton>
          </div>
        </div>
      ) : viewAppLink ? (
        <div className="InformationBarWithButton">
          <div className="InfoHeader">{header}</div>
          <a href={viewAppLink} rel="noopener noreferrer" target="_blank">
            <PrimaryButton color="primary-outline">Open App</PrimaryButton>
          </a>
        </div>
      ) : credits ? (
        <div className="InformationBarWithButton">
          <div className="InfoHeader">{header}</div>
          {/**appears if user has credits */}
          <div className="CreditsCorner" title="credits">
            {credits > 0 ? credits : 0}
            <Coin />
          </div>
        </div>
      ) : (
        <div className="InfoHeader">{header}</div>
      )}
    </div>
  );
};

export default InformationBar;
