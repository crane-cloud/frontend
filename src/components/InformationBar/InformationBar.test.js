import React from "react";
import { shallow, mount } from "enzyme";
import InformationBar from ".";

describe("InformationBar Component", () => {
  const searchAction = jest.fn();
  const btnAction = jest.fn();

  const defaultProps = {
    header: "test",
    status: undefined,
    showBtn: false,
    btnAction: btnAction,
    viewAppLink: undefined,
    showSearchBar: false,
    placeholder: undefined,
    searchAction: searchAction,
  };

  it("renders the default information bar without errors", () => {
    const wrapper = render(<InformationBar {...defaultProps} />);
    expect(wrapper.exists()).toBe(true);
  });

  it("renders the information bar with search bar and triggers search action", () => {
    const searchBarProps = {
      ...defaultProps,
      showSearchBar: true,
      placeholder: "test",
    };
    const wrapper = render(<InformationBar {...searchBarProps} />);

    // const searchInput = wrapper.find('.DesktopView .SearchInput input');
    // searchInput.simulate('change', { target: { name: 'Searchword', value: 'sample search' } });

    // expect(searchAction).toHaveBeenCalledWith('sample search');
  });

  it("renders the information bar with app status", () => {
    const appStatusProps = {
      ...defaultProps,
      status: "success",
    };
    const wrapper = render(<InformationBar {...appStatusProps} />);

    expect(wrapper.find("AppStatus")).toHaveLength(1);
    expect(wrapper.find(".AppUrl a")).toHaveLength(1);
  });

  it("renders the information bar with a button and triggers button action", () => {
    const addBtnProps = {
      ...defaultProps,
      showBtn: true,
    };
    const wrapper = render(<InformationBar {...addBtnProps} />);

    expect(wrapper.find("PrimaryButton")).toHaveLength(1);
    wrapper.find("PrimaryButton").simulate("click");
    expect(btnAction).toHaveBeenCalled();
  });

  it("renders the information bar with view app link", () => {
    const addAppLinkProps = {
      ...defaultProps,
      viewAppLink: "test_url",
    };
    const wrapper = render(<InformationBar {...addAppLinkProps} />);

    expect(wrapper.find(".InfoHeader")).toHaveLength(1);
    expect(wrapper.find(".InfoHeader").text()).toBe("test");
    expect(wrapper.find(".PrimaryButton")).toHaveLength(0);
    // expect(wrapper.find('.PrimaryButton').prop('children')).toBe("Open App");
    expect(wrapper.find("a")).toHaveLength(1);
    expect(wrapper.find("a").prop("href")).toBe("test_url");
  });

  it("renders the information bar with credits", () => {
    const creditsProps = {
      ...defaultProps,
      credits: 10,
    };
    const wrapper = render(<InformationBar {...creditsProps} />);

    expect(wrapper.find(".InfoHeader")).toHaveLength(1);
    expect(wrapper.find(".InfoHeader").text()).toBe("test");
    expect(wrapper.find(".CreditsCorner")).toHaveLength(1);
    expect(wrapper.find(".CreditsCorner").text()).toBe("10");
    expect(wrapper.find("Coin")).toHaveLength(0);
  });

  it("renders the information bar without additional elements", () => {
    const wrapper = render(<InformationBar {...defaultProps} />);

    expect(wrapper.find(".InfoHeader")).toHaveLength(1);
    expect(wrapper.find(".InfoHeader").text()).toBe("test");
    expect(wrapper.find(".InformationBarWithButton")).toHaveLength(0);
    expect(wrapper.find(".AppUrl")).toHaveLength(0);
    expect(wrapper.find(".RoundAddButtonWrap")).toHaveLength(0);
    expect(wrapper.find("PrimaryButton")).toHaveLength(0);
    expect(wrapper.find("a")).toHaveLength(0);
    expect(wrapper.find(".CreditsCorner")).toHaveLength(0);
    expect(wrapper.find("Coin")).toHaveLength(0);
  });
});
