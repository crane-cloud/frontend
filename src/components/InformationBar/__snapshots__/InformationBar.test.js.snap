// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationBar Component renders the default information bar without errors 1`] = `ShallowWrapper {}`;
